### ISSUE TITLE
Allow InteractionOptionResolvers to return the raw value for different option types (like user)

### ISSUE THREAD

**Issue by vladfrangu** [discordjs/discord.js#10344]
### Which application or package is this feature request for?

discord.js

### Feature

Messing around with autocomplete interactions, I've found out that we don't have a _nicer_ way of getting the raw id provided in a previous option when autocompleting outside of doing `interaction.options.get()`. Unfortunately, we also cannot provide the full methods because Discord doesn't send us resolved data in this interaction type, only the ids.

### Ideal solution or implementation

Ideally, we'd provide shortcuts for getting those ids, something like `getUserValue` or some much better name than what I came up with on the spot!

### Other context

```ts
new SlashCommandBuilder()
	.setName('example')
	.setDescription('example')
	.addUserOption((user) =>
		user.setName('user').setDescription('The user to filter autocomplete by').setRequired(true),
	)
	.addStringOption((id) =>
		id
			.setName('autofill-me')
			.setDescription('Value that gets autofilled with data dependent on whatever user has, or nothing is user is not set')
			.setRequired(true)
			.setAutocomplete(true),
	);
```

**Comment by Jiralite** [comment]
What about making the second parameter of all these methods an object that has a `raw` option (and `required`, of course) to return the payload from the API?

**Comment by Qjuh** [comment]
Isn’t that what `<AutocompleteInteraction>.options.get(…)` already gives you?

**Comment by Jiralite** [comment]
Yeah... from reading the feature request, using that method is not desired. Unless I'm reading it wrong.

**Comment by monbrey** [comment]
Is this just so the raw value can be correctly typed between string, number and boolean? There's otherwise no benefit to it being a "getUserValue" method as all Snowflakes are strings.
